### PR TITLE
Register accounts on deposit and enable registration with MetaMask.

### PR DESCRIPTION
**Pull request by nepoche**
## Summary of changes
- Require a NoteAccount to complete the Deposit flow on the DApp.
- Enable a button for 'Connecting With Metamask'.

### Reference issue to close (if applicable)

-----
### Code Checklist 

- [x] Tested
- [ ] Documented

https://user-images.githubusercontent.com/16654909/185417806-813d259c-56eb-4c73-9ecd-a4bef0a108a4.mov

### CODE REVIEW

**Comment by dutterbutter** [comment]
@nepoche I seem to be running into an issue regarding loading notes from the account manager. To reproduce:
1. Create a note account
2. Paste a note into the load note input box 
3. The page then sends me to the following:
Screen Shot 2022-08-18 at 12 33 07 PM

Furthermore, if I reload the UI and select the note account manager I am redirected back to the above error message page.

**Comment by dutterbutter** [comment]
> @nepoche I seem to be running into an issue regarding loading notes from the account manager. To reproduce:
> 
> 1. Create a note account
> 2. Paste a note into the load note input box
> 3. The page then sends me to the following:
> 
> Screen Shot 2022-08-18 at 12 33 07 PM
> 
> Furthermore, if I reload the UI and select the note account manager I am redirected back to the above error message page.

**Update:** It seems it broke using a very old note, new notes generated works fine.

**Review by AhmedKorim — APPROVED** [review]
LGTM!

**Comment by drewstone** [comment]
@AhmedKorim I can't tell if you ran into the issues @dutterbutter is facing but it would be good to QA these PRs certainly when others are facing issues. LGTM is not enough here.

File: packages/api-providers/src/web3/vanchor-registration.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { VAnchorRegistration } from '../abstracts/anchor/vanchor-registration';
+import { registrationStorageFactory } from '../utils';
+import { WebbWeb3Provider } from './webb-provider';
+
+export class Web3VAnchorRegistration extends VAnchorRegistration<WebbWeb3Provider> {
+  // Check if the evm address and public key pairing has already registered.
+  async isPairingRegistered(anchorAddress: string, account: string, pubkey: string): Promise<boolean> {
```
Comment (drewstone) [inline packages/api-providers/src/web3/vanchor-registration.ts:7]: I would consider renaming to `isPairRegistered`

**Review by drewstone — APPROVED** [review]
LGTM, just small renaming thought, feel free to shut down if you think it's better named as is.

**Review by drewstone — CHANGES_REQUESTED** [review]
Code looks good, let's do more thorough QA to ensure @dutterbutter's issues are resolved.

**Comment by dutterbutter** [comment]
It looks like my issue has been resolved, as I am unable to reproduce now.

**Pull request merged**